IceCreamShop.playground/Contents.swift

```diff
@@ -1 +1,83 @@
+import UIKit
+
+struct Flavor {
+    var name: String
+    var rating: Double
+}
+
+enum Size: Double {
+    case small = 1.00
+    case medium = 2.88
+    case large = 5.25
+}
+
+struct Cone {
+    let flavor: Flavor
+    let topping: String?
+    let size: Size
+    var totalSales: Double = 0.0
+    
+    func eat() {
+        print("Yum, I love \(flavor.name)!)")
+    }
+}
+
+class IceCreamShop {
+    var flavors: [Flavor]
+    var sizes: [Size]
+    var toppings: [String?]
+
+    
+    init(flavors: [Flavor], sizes: [Size], toppings: [String?]) {
+        self.flavors = flavors
+        self.sizes = sizes
+        self.toppings = toppings
+ //       self.totalSales = totalSales
+    }
+}
+
+
+func listTopFlavors(_ flavors: [Flavor]) {
+    var flavors: [Flavor] = flavors
+    
+    for flavor in flavors {
+        if flavor.rating >= 4.0 {
+            flavors.append(contentsOf: flavors)
+            print("Our top flavors are \(flavor.name)!")
+        }
+    }
+}
```

Comment:
Look at putting this function, as well as the order cone one inside of your ice cream shop.